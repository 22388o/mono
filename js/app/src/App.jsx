import { useState, useSyncExternalStore } from 'react'
import {
  Route,
  BrowserRouter as Router,
  Routes
} from 'react-router-dom'
import { Buffer } from 'buffer';
import { AppContext } from '../context'
import { ThemeProvider } from '@mui/material';
import { DEFAULT_THEME } from './utils/constants';
import { SwapHome } from './components/SwapHome';
import './App.css';
import './index.css';
import { Sdk } from '@portaldefi/sdk';
import { sdkStore } from './syncstore/sdkstore';

if(typeof window !== 'undefined') {
  (window).Buffer = Buffer;
}

function App () {
  const SDK = useSyncExternalStore(sdkStore.subscribe, () => sdkStore.currentState);
  const [context, setContext] = useState({})
  const url = new URL(window.location);
  const sdkProps = {
    id: 'bob',
    hostname: url.hostname,
    port: Number(url.port),
    pathname: '/api/v1'
  }
  const sdk = new Sdk(sdkProps)
    .on("log", (level, ...args) => console[level](...args))
    .on("error", (err, ...args) => console.error(err, ...args))
  sdk.start().then(()=>{
    await sdkStore.dispatch({ type: 'SIGN_IN', payload: sdk })
  })
  
  return (
    <ThemeProvider theme={DEFAULT_THEME}>
      <AppContext.Provider value={{ context, setContext }}>
        <Router>
          <Routes>
            <Route path='' element={<SwapHome />} />
          </Routes>
        </Router>
      </AppContext.Provider>
    </ThemeProvider>
  )
}

export default App
