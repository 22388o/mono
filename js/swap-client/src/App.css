#root {
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.card, .cards, .content {
  border-radius: 16px;
  color: #EAEAEA;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

.ui.button {
  /* bw/EA */
  background: #EAEAEA;
  /* bw/EA */
  border: 1px solid #EAEAEA !important;
  border-radius: 16px;

}

.ui.primary.button {
  /* util/brand-gradient */
  background: radial-gradient(100% 181.89% at 0% 100%, #748AFE 7.29%, #A64EFF 100%) /* warning: gradient uses a rotation that is not supported by CSS and may not behave as expected */;
  border-radius: 16px;
}

.ui.inline.dropdown {
  padding-top:0.3em;
  font-size: 1.5em;
  color: #000;
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.user-swap {
  width: 50% !important;
  float: left !important;
}

body {
  background-color: #191b21 !important;
  color: white !important;
}

.flex-nowrap { flex-wrap: nowrap !important; }

.sw-title { width: 100px !important;}

td { overflow-wrap: anywhere;}

table { background-color: #191b21 !important; }

label, .header { color: white !important; }

button { padding: 0.5rem !important }

.card { background-color: black !important; }
.card * { color: white; }

h3 {
  font-size: 1.28571429rem !important;
  font-weight: bold !important;
}

.divider { margin: 0 !important; }

.left { text-align: left !important; }
.right { text-align: right !important; }
