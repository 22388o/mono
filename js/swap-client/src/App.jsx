import { useState } from 'react'
import { Provider } from 'react-redux';
import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import 'semantic-ui-css/semantic.min.css';
import './App.css'
import { Swap } from './components/Swap';
import { SwapHistory } from './components/SwapHistory';
import { store } from './store';

function App() {
  const [count, setCount] = useState(0)

  return (
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path='' element={<Swap />} />
          <Route path='/history' element={<SwapHistory />} />
        </Routes>
      </Router>
    </Provider>
  )
}

export default App
