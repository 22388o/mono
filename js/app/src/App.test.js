import { create, act } from 'react-test-renderer'
import App from './App';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ data: { amount: 12 } }),
  })
);

describe('App test', () => {
  it('should work', () => {
    let tree

    act(() => {
      tree = create(
        <App />
      )
    })
    expect(tree).toMatchSnapshot()

  })
  afterAll(() => jest.resetModules())
})
