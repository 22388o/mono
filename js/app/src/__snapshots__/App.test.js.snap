// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should work 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-uz3bcw-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item css-hoc6b0-MuiGrid-root"
    style={
      {
        "padding": "1em 2em",
      }
    }
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10 css-1dsmj4t-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-0"
      >
        <div
          className="MuiStack-root css-163ljxj-MuiStack-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1o9533o-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-z58hwb-MuiGrid-root"
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <h3>
                Swap
              </h3>
            </div>
          </div>
          <div
            className="MuiGrid-root css-vj1n65-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1tj2tdk-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
              >
                <input
                  className="undefined qty-input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder={1000}
                  type="number"
                  value=""
                />
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-nyvysi-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <img
                    src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true"
                  />
                  BTC
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              style={
                {
                  "borderColor": "#202020",
                  "margin": "0.5em -1em 0 -1em",
                }
              }
            />
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium undefined exchange css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SettingsEthernetIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.77 6.76 6.23 5.48.82 12l5.41 6.52 1.54-1.28L3.42 12l4.35-5.24zM7 13h2v-2H7v2zm10-2h-2v2h2v-2zm-6 2h2v-2h-2v2zm6.77-7.52-1.54 1.28L20.58 12l-4.35 5.24 1.54 1.28L23.18 12l-5.41-6.52z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              className="MuiGrid-root MuiGrid-container mt-m1 mb-0 css-1tj2tdk-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
              >
                <input
                  className="undefined qty-input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder={1000}
                  type="number"
                  value=""
                />
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-nyvysi-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <img
                    src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
                  />
                  ETH
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root css-e54n9e-MuiGrid-root"
          >
            <button
              circular="true"
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium w-100 h-3 gradient-btn-disabled css-nyvysi-MuiButtonBase-root-MuiButton-root"
              disabled={true}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              secondary="true"
              tabIndex={-1}
              type="button"
            >
              Enter Amounts to Swap
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    style={
      {
        "display": "block",
      }
    }
  >
    <div
      className="MuiStack-root css-1rzuxdl-MuiStack-root"
    >
      <div
        className="MuiStack-root css-m69qwo-MuiStack-root"
      >
        <div
          className="MuiStack-root css-313u06-MuiStack-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 undefined undefined css-1tet1fr-MuiTypography-root"
            onClick={[Function]}
          >
            Activity
          </p>
        </div>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardDoubleArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.41 6 5 7.41 9.58 12 5 16.59 6.41 18l6-6z"
            />
            <path
              d="m13 6-1.41 1.41L16.17 12l-4.58 4.59L13 18l6-6z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiStack-root css-nen11g-MuiStack-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-3v0wvh-MuiTypography-root"
        >
          No Activities
        </p>
      </div>
    </div>
  </div>
</div>
`;
